libs/block-editor/src/lib/extensions/services/dot-language/dot-language.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+
+import { Observable, of } from 'rxjs';
+import { map, pluck } from 'rxjs/operators';
+
+// eslint-disable-next-line max-len
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+interface Language {
+    country: string;
+    countryCode: string;
+    defaultLanguage: boolean;
+    id: number;
+    language: string;
+    languageCode: string;
+}
+
+export interface DotLanguage {
+    [key: string]: Language;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DotLanguageService {
+    public language: DotLanguage;
+
+    constructor(private http: HttpClient) {}
+
+    get defaultHeaders() {
+        const headers = new HttpHeaders();
+        headers.set('Accept', '*/*').set('Content-Type', 'application/json');
+        return headers;
+    }
+
+    getLanguages(): Observable<DotLanguage> {
+        if (this.language) {
+            return of(this.language);
+        }
+
+        return this.http
+            .get(`/api/v2/languages`, {
+                headers: this.defaultHeaders
+            })
+            .pipe(
+                pluck('entity'),
+                map((lang: Language[]) => {
+                    const dotLang: DotLanguage = lang.reduce(
+                        (obj, lang) => Object.assign(obj, { [lang.id]: lang }),
+                        {}
+                    );
+
+                    this.language = dotLang;
+
+                    return dotLang;
+                })
+            );
+    }
+}
```

Comment:
This looks like a service for a public library... don't we have one already?

Reply:
We have one [`apps/dotcms-ui/src/app/api/services/dot-languages/dot-languages.service.ts`] that we use to get the languages. The problem is that this service needs CoreWebService and that one has other dependencies, so I don't think it's worth using it in the editor to get only the languages, we're going to have to import several services in the editor and make it heavier.